benchmark/distinct_user_id/rapidjson_sax.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#ifdef SIMDJSON_COMPETITION_RAPIDJSON
+
+#include "distinct_user_id.h"
+#include <string.h>
+namespace distinct_user_id {
+
+using namespace rapidjson;
+
+struct rapidjson_sax {
+    struct Handler {
+        std::vector<uint64_t>* result;
+        bool user = false;
+        bool user_id =false;
+        Handler(std::vector<uint64_t> &r) { result=&r; }
+
+        bool Key(const char* key, SizeType length, bool copy) {
+            if (strcmp(key,"user") == 0) { user = true; } // Checking if entering user object
+            else if (user && strcmp(key,"id") == 0) { user_id = true; } // Checking if in an user object and accessing id field
+            return true;
+        }
+        bool Uint(unsigned i) {
+            if (user_id) {  // Getting id if previous key was "id" for a user
+                (*result).emplace_back(i);
+                user_id=false;
+                user = false;
+            }
+            return true;
+        }
+        // Irrelevant events
+        bool Null() { return true; }
+        bool Bool(bool b) { return true; }
+        bool Double(double d) { return true; }
+        bool Int(int i) { return true; }
+        bool Int64(int64_t i) { return true; }
+        bool Uint64(uint64_t i) { return true; }
```

Comment:
In the DOM code, we call `result.push_back(id->value.GetUint64())` which would suggest that the IDs are treated as  `uint64_t` values.

Here is what the documentation states:

> And then it calls one function out of Int(int), Uint(unsigned), Int64(int64_t), Uint64(uint64_t) and Double(double).

It is entirely possible that this code is correct, but I suggest you document it briefly to avoid confusing.